api/primary_test.go

```diff
@@ -56,8 +57,43 @@ func TestCorsRequest(t *testing.T) {
 		t.Log(k, " : ", v)
 	}
 
-	if w.Code == 404 {
-		t.Fatalf("failed not found")
+	// test response status code
+	if w.Code != 200 {
+		t.Fatalf("HTTP response status code is %d, not 200", w.Code)
+	}
+
+	// test Access-Control-Allow-Headers in response headers
+	// allowHeaders has the type of slice with only one string element
+	allowHeaders, exist := w.Header()["Access-Control-Allow-Headers"]
+	if !exist {
+		t.Fatal("no Access-Control-Allow-Headers in response")
+	}
+
+	if len(allowHeaders) != 1 {
+		t.Fatal("length of Access-Control-Allow-Headers is not 1")
+	}
+
+	for _, value := range []string{"Origin", "X-Requested-With", "Content-Type", "Accept", "X-Registry-Auth"} {
+		if !strings.Contains(allowHeaders[0], value) {
+			t.Fatalf("Header %s is not in Access-Control-Allow-Headers", value)
+		}
+	}
+
+	// test Access-Control-Allow-Methods in response headers
+	// allowMethods has the type of slice with only one string element
+	allowMethods, exist := w.Header()["Access-Control-Allow-Methods"]
+	if !exist {
+		t.Fatal("no Access-Control-Allow-Methods in response")
+	}
+
+	if len(allowMethods) != 1 {
```

Comment:
same as above if the above made sense. Probably don't need the comment the second time, though. It is now a pattern.